src/main/scala/is/hail/asm4s/Code.scala

```diff
@@ -353,6 +353,9 @@ class CodeBoolean(val lhs: Code[Boolean]) extends AnyVal {
   def ||(rhs: Code[Boolean]): Code[Boolean] =
     lhs.toConditional || rhs.toConditional
 
+  def ^(rhs: Code[Boolean]): Code[Boolean] =
```

Comment:
mild suggestion: group the conditionals and then the bitwise, so  &&, || then &, |, ^

Reply:
None of these are bit-wise operators, they all operate on booleans. `&&` is lazy-and and `&` is strict-and. `||` is lazy-or and `|` is strict-or. `^` is strict-xor.